huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_consumer_group.go

```diff
@@ -0,0 +1,300 @@
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the list of associated brokers of the consumer group.`,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the name of the consumer group.`,
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexp.MustCompile(`^[A-Za-z|%-_0-9]*$`),
+						"An instance name starts with a letter and can contain only letters, digits,"+
+							"vertical lines(|), percent sign(%), underscores (_), and hyphens (-)"),
+					validation.StringLenBetween(3, 64),
+				),
+			},
+			"retry_max_time": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				Description:  `Specifies the maximum number of retries.`,
+				ValidateFunc: validation.IntBetween(1, 16),
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Computed:    true,
+				Description: `Specifies the consumer group is enabled or not.`,
+			},
+		},
+	}
+}
+
+func resourceDmsRocketMQConsumerGroupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+
+	// createRocketmqConsumerGroup: create DMS rocketmq consumer group
+	var (
+		createRocketmqConsumerGroupHttpUrl = "v2/{project_id}/instances/{instance_id}/groups"
+		createRocketmqConsumerGroupProduct = "dms"
+	)
+	createRocketmqConsumerGroupClient, err := config.NewServiceClient(createRocketmqConsumerGroupProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DmsRocketMQConsumerGroup Client: %s", err)
+	}
+
+	instanceID := d.Get("instance_id").(string)
+	createRocketmqConsumerGroupPath := createRocketmqConsumerGroupClient.Endpoint + createRocketmqConsumerGroupHttpUrl
+	createRocketmqConsumerGroupPath = strings.ReplaceAll(createRocketmqConsumerGroupPath, "{project_id}", createRocketmqConsumerGroupClient.ProjectID)
+	createRocketmqConsumerGroupPath = strings.ReplaceAll(createRocketmqConsumerGroupPath, "{instance_id}", instanceID)
+
+	createRocketmqConsumerGroupOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	createRocketmqConsumerGroupOpt.JSONBody = utils.RemoveNil(buildCreateRocketmqConsumerGroupBodyParams(d, config))
+	createRocketmqConsumerGroupResp, err := createRocketmqConsumerGroupClient.Request("POST", createRocketmqConsumerGroupPath, &createRocketmqConsumerGroupOpt)
+	if err != nil {
+		return diag.Errorf("error creating DmsRocketMQConsumerGroup: %s", err)
+	}
+
+	createRocketmqConsumerGroupRespBody, err := utils.FlattenResponse(createRocketmqConsumerGroupResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("name", createRocketmqConsumerGroupRespBody)
+	if err != nil {
+		return diag.Errorf("error creating DmsRocketMQConsumerGroup: ID is not found in API response")
```

Comment:
need to return "err" message.

Reply:
this err is not found error, for this kind error, this err info will be returned